packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Snippets/BinaryDataSnippets.cs

```diff
@@ -22,7 +22,7 @@ public static ScopedApi<BinaryData> FromStream(ValueExpression stream, bool asyn
             return Static<BinaryData>().Invoke(methodName, [stream], null, callAsAsync: async).As<BinaryData>();
         }
 
-        public static ValueExpression ToMemory(this ScopedApi<BinaryData> binaryData) => binaryData.Invoke(nameof(BinaryData.ToMemory));
+        public static ScopedApi<ReadOnlyMemory<byte>> ToMemory(this ScopedApi<BinaryData> binaryData) => new(binaryData.Invoke(nameof(BinaryData.ToMemory)));
```

Comment:
Use .As insead of new.  This avoids duplicate allocations within a single expression.  If the underlying invoke ever got more intelligent and returned a scopedapi this would not take advantage of that.

In general the ctor for scopedapi should only be called from the As methods this is a good candidate for an analyzer